src/osd/OSD.cc

```diff
@@ -6584,12 +6584,9 @@ struct C_OnMapCommit : public Context {
 
 struct C_OnMapApply : public Context {
   OSDService *service;
-  list<OSDMapRef> pinned_maps;
```

Comment:
This is actually really important.  As long as the refs are alive, the OSDMaps will be pinned in the cache and we won't try to read it off of disk.  If you remove this, the maps will not stay in the cache, and we will try to read those OSDMaps before they are actually written resulting in a crash.  You should remove this commit.